### ISSUE TITLE
Excessively verbose error when update fails

### ISSUE THREAD

**Issue by erictune** [GoogleCloudPlatform/kubernetes#9721]
What happened:
```
$ kubectl update -f gh-tag.yaml
The Pod "gh-tag" is invalid:spec: spec: invalid value '{Volumes:[{Name:secret-volume VolumeSource:
{HostPath:<nil> EmptyDir:<nil> GCEPersistentDisk:<nil> AWSElasticBlockStore:<nil> GitRepo:<nil>
 Secret:<*>(0xc20a477cc0){SecretName:gh-secret} NFS:<nil> ISCSI:<nil> Glusterfs:<nil> 
PersistentVolumeClaimVolumeSource:<nil> RBD:<nil>}} {Name:default-token-04gjq VolumeSource:
{HostPath:<nil> EmptyDir:<nil> GCEPersistentDisk:<nil> AWSElasticBlockStore:<nil> GitRepo:<nil> 
Secret:<*>(0xc20a477cf0){SecretName:default-token-04gjq} NFS:<nil> ISCSI:<nil> Glusterfs:<nil> 
PersistentVolumeClaimVolumeSource:<nil> RBD:<nil>}}] Containers:[{Name:test-container 
Image:progrium/busybox Command:[sh -c mkdir /tmp/git; which git; cd /tmp/git; pwd; which git; ] Args:
<nil> WorkingDir: Ports:<nil> Env:<nil> Resources:{Limits:map[cpu:100m] Requests:map[]} 
VolumeMounts:[{Name:secret-volume ReadOnly:false MountPath:/etc/secret-volume} {Name:default-
token-04gjq ReadOnly:true MountPath:/var/run/secrets/kubernetes.io/serviceaccount}] LivenessProbe:
<nil> ReadinessProbe:<nil> Lifecycle:<nil> TerminationMessagePath:/dev/termination-log 
ImagePullPolicy:IfNotPresent SecurityContext:<*>(0xc20964e020){Capabilities:<*>(0xc20a0cd080){Add:
<nil> Drop:<nil>} Privileged:<*>(0xc20a477c3c)false SELinuxOptions:<nil> RunAsUser:<nil>}}] 
RestartPolicy:Never TerminationGracePeriodSeconds:<nil> ActiveDeadlineSeconds:<nil> 
DNSPolicy:ClusterFirst NodeSelector:<nil> ServiceAccount:foo NodeName:kubernetes-minion-z282 
HostNetwork:false ImagePullSecrets:<nil>}': may not update fields other than container.image
```

I don't need to see addresses of pointers of the internal representation.  And I don't even need to see any representation of the object, since the last part of the error message is pretty clear.

What I would like to have seen:
``` 
Update failed: may not update fields other than container.image
```

Ideally, it would also say which fields it thinks I am changing.  But just removing the spew would be a net improvement.

**Comment by satnam6502** [comment]
For reference, current output:
```
$ kubectl update -f counter-pod.yaml 
The Pod "counter" is invalid:spec: spec: invalid value '{Volumes:<nil> Containers:[{Name:count Image:ubuntu:14.04 Command:<nil> Args:[bash -c for ((i = 0; ; i++)); do echo "$i: $(date)"; sleep 1; done] WorkingDir: Ports:<nil> Env:<nil> Resources:{Limits:map[cpu:100m] Requests:map[]} VolumeMounts:<nil> LivenessProbe:<nil> ReadinessProbe:<nil> Lifecycle:<nil> TerminationMessagePath:/dev/termination-log ImagePullPolicy:IfNotPresent SecurityContext:<nil>}] RestartPolicy:Always TerminationGracePeriodSeconds:<nil> ActiveDeadlineSeconds:<nil> DNSPolicy:ClusterFirst NodeSelector:<nil> ServiceAccount: NodeName: HostNetwork:false ImagePullSecrets:<nil>}': may not update fields other than container.image
```

**Comment by satnam6502** [comment]
The changes in PR change the output to:
```
$ kubectl update -f counter-pod.yaml 
Update failed:  may not update fields other than container.image
```

**Comment by satnam6502** [comment]
Closed via #9803
